src/borg/_chunker.c

```diff
@@ -335,6 +335,12 @@ chunker_is_zero(const uint8_t *data, size_t length)
 }
 #endif
 
+/*
+ * performance (Haswell, E3-1231v3)
+ * without chunker_is_zero (SSE), about 710-720 MB/s
+ * with    "                "   , about 690-700 MB/s
```

Comment:
i did some practical testing and I can't see a sizeable difference in pure dedup scenarios between HEAD and this branch, so it seems that the lenmod change offsets this additional check completely.

There is btw. another enhancing change in that region, by hardcoding lenmod = 31 (4095 & 0x1f) I get to 750-760 MB/s in this branch, but it naturally breaks the tests, which use a window size < 4095. I think the reason might be that x86 has a separate coding for `rol` where the count is immediate, not through a register, which seems to be faster.